src/Scope-RedPitaya_STEMlab/main.py

```
@@ -65,226 +66,483 @@ def __init__(self):
+            self.triggertimeout = float(parameters.get("TriggerTimeout", 5.0))
+            self.triggerhysteresis = float(parameters.get("TriggerHysteresis", 0.0))
+            self.timeoffsetvalue = float(parameters.get("TimeOffsetValue", 0.0))
+            self.averages = int(parameters.get("Average", 1))
+            self.samples = int(parameters.get("SamplingRate", 16384))
+            self.channel_offsets = {
+                i: float(parameters.get(f"Channel{i}_Offset", 0.0))
+                for i in self.channels
+            }
+        except ValueError:
+            msg = "Red Pitaya STEMlab: Please enter numeric values into all number fields."
+            raise ValueError(msg)
```

Ruff (0.16.1)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)